docs/authentication.md

```diff
@@ -1,52 +1,145 @@
-instance, we can easily authenticate the bot in OBO mode:
+Configuring both RSA and certificate authentication isn't recommended. 
 
+### Bot authentication deep-dive
+The code snippet below explains how to manually retrieve your bot authentication session. However, note that by default 
+those operations are done behind the scene through the `SymphpnyBdk` entry point.
 ```java
 public class Example {
-    
-    public static void main(String[] args) {
-        // Initialize the BDK entry point
+
+    public static void main(String[] args) throws Exception { 
+        // create bdk entry point
         final SymphonyBdk bdk = new SymphonyBdk(loadFromClasspath("/config.yaml"));
-        // OBO authentication
-        AuthSession oboSession = bdk.obo("user.name");
-        // Running service in Obo mode
-        final List<V2UserDetail> userDetails = bdk.users().listUsersDetail(new UserFilter(), oboSession);
+        // at this point your bot is already authenticated
+        // here's how to retrieve the authentication session
+        final AuthSession botSession = bdk.botSession();
+        log.info("sessionToken: {}", botSession.getSessionToken());
+        log.info("keyManagerToken: {}", botSession.getKeyManagerToken());
+        // if session has expired (e.g. an API endpoint returns 401), you can manually trigger a re-auth
+        botSession.refresh();
+    }
+}
+```
+
+### Multiple bot instances
+By design, the `SymphonyBdk` object contains a single bot session. However, you might want to create an application that
+has to handle multiple bot sessions, potentially using different authentication modes. This is possible by creating 
+multiple instances of `SymphpnyBdk` using different configurations:
+```java
+public class Example {
+
+    public static void main(String[] args) throws Exception { 
+
+        final SymphonyBdk bot1 = new SymphonyBdk(loadFromClasspath("/config-bot1.yaml"));
+        final SymphonyBdk bot2 = new SymphonyBdk(loadFromClasspath("/config-bot2.yaml"));
+        
+        // bot2 creates an IM with bot1
+        bot2.streams().create(bot1.botInfo().getId());
     }
 }
+```
 
+## App authentication
+Application authentication is completely optional but remains required if you want to implement the Circle Of trust 
+or if you want to use OBO.
+
+### App authentication using RSA
+
+Required `config.yaml` setup: 
+```yaml
+host: acme.symphony.com
+app:
+    appId: app-id
+    privateKeyPath: /path/to/rsa/private-key.pem
 ```
 
-## Extension app authentication
+### App Authentication using Client Certificate
 
-In order to authenticate your extension app, you need to have a proper `app` field defined in your configuration,
-see [authentication](authentication.md) for more information. We support both RSA and certificate authentication.
+Required `config.yaml` setup: 
+```yaml
+host: acme.symphony.com
+app:
+    appId: app-id
+    certificatePath: /path/to/certificate.cer
```

Comment:
The certificate for extension app should be in pem format.
https://developers.symphony.com/extension/docs/application-authentication#section-prerequisites-for-client-certificate-authentication

Reply:
IMHO documentation is not so clear. We use PEM format when registering the public certificate in the admin console. However, we use a p12 file which is in DER format when authenticating.

Reply:
@symphony-elias no, I used the crt format when registering the public certificate in the admin console and the p12 when authenticating.